src/dos/dos_keyboard_layout.cpp

```diff
@@ -1243,6 +1244,164 @@ const char* DOS_GetLoadedLayout(void) {
+	{"mt103",  Country::Malta          }, // US-based
+	{"ne",     Country::Niger          },
+	{"ng",     Country::Nigeria        },
+	{"nl",     Country::Netherlands    }, // 102-key
+	{"no",     Country::Norway         },
+	{"ph",     Country::Philippines    },
+	{"pl",     Country::Poland         }, // 101-key, Programmers
+	{"pl214",  Country::Poland         }, // 102-key
+	{"po",     Country::Portugal       },
+	{"px",     Country::Portugal       }, // International
+	{"ro",     Country::Romania        }, // Standard
+	{"ro446",  Country::Romania        }, // QWERTY
+	{"rh",     Country::United_States  }, // Right-Hand Dvorak
+	{"ru",     Country::Russia         }, // Standard
+	{"ru443",  Country::Russia         }, // Typewriter
+	{"rx",     Country::Russia         }, // Extended Standard
+	{"rx443",  Country::Russia         }, // Extended Typewriter
+	{"sd",     Country::Switzerland    }, // German
+	{"sf",     Country::Switzerland    }, // French
+	{"sg",     Country::Switzerland    }, // German
+	{"si",     Country::Slovenia       },
+	{"sk",     Country::Czech_Slovak   }, // Slovakia
+	{"sp",     Country::Spain          },
+	{"sq",     Country::Albania        }, // No-deadkeys
+	{"sq448",  Country::Albania        }, // Deadkeys
+	{"sr",     Country::Serbia         }, // Deadkey
+	{"su",     Country::Finland        },
+	{"sv",     Country::Sweden         },
+	{"sx",     Country::Spain          }, // International
+	{"tj",     Country::Tadjikistan    },
+	{"tm",     Country::Turkmenistan   },
+	{"tr",     Country::Turkey         }, // QWERTY
+	{"tr440",  Country::Turkey         }, // Non-standard
+	{"tt",     Country::Russia         }, // Tatarstan Standard
+	{"tt443",  Country::Russia         }, // Tatarstan Typewriter
+	{"ua",     Country::Ukraine        }, // 101-key
+	{"uk",     Country::United_Kingdom }, // Standard
+	{"uk168",  Country::United_Kingdom }, // Allternate
+	{"ur",     Country::Ukraine        }, // 101-key
+	{"ur465",  Country::Ukraine        }, // 101-key
+	{"ur1996", Country::Ukraine        }, // 101-key
+	{"ur2001", Country::Ukraine        }, // 102-key
+	{"ur2007", Country::Ukraine        }, // 102-key
+	{"us",     Country::United_States  }, // Standard
+	{"ux",     Country::United_States  }, // International
+	{"uz",     Country::Uzbekistan     },
+	{"vi",     Country::Vietnam        },
+	{"yc",     Country::Serbia         }, // Deadkey
+	{"yc450",  Country::Serbia         }, // No-deadkey
+	{"yu",     Country::Yugoslavia     },
+        // clang-format on
+};
+
+static bool country_number_exists(const int requested_number)
+{
+	for ([[maybe_unused]] const auto &[code, number] : country_code_map)
+		if (requested_number == static_cast<int>(number))
+			return true;
+	return false;
+}
+
+static Country lookup_country_from_code(const char *country_code)
+{
+	if (country_code)
+		if (const auto it = country_code_map.find(country_code);
+		    it != country_code_map.end())
+			return it->second;
```

Comment:
Minor: wouldn't it be easier to see what's going on with added braces around the statements?